File: src/chapters/1_fundamental_data_structures/pascals_triangle/pascals_triangle.cpp
#include <iostream>
#include <vector>
using namespace std;

void printPascalsTriangle(int n) {
    vector<vector<int>> triangle(n);

    for (int i = 0; i < n; i++) {
        triangle[i].resize(i + 1);
        triangle[i][0] = triangle[i][i] = 1;

        for (int j = 1; j < i; j++) {
            triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
        }
    }

    for (int i = 0; i < n; i++) {
        for (int j = 0; j <= i; j++) {
            cout << triangle[i][j] << " ";
        }
        cout << "\n";
    }
}

int main() {
    int rows;
    cout << "Enter number of rows: ";
    cin >> rows;
    printPascalsTriangle(rows);
    return 0;
}

File: src/chapters/1_fundamental_data_structures/pascals_triangle/pascals_triangle.pas
program PascalsTriangle;

var
  triangle: array[1..20, 1..20] of integer;
  i, j, n: integer;

begin
  write('Enter number of rows: ');
  readln(n);

  for i := 1 to n do
  begin
    triangle[i, 1] := 1;
    triangle[i, i] := 1;

    for j := 2 to i - 1 do
      triangle[i, j] := triangle[i - 1, j - 1] + triangle[i - 1, j];
  end;

  writeln('Pascal''s Triangle:');
  for i := 1 to n do
  begin
    for j := 1 to i do
      write(triangle[i, j], ' ');
    writeln;
  end;
end.

